transport-native-epoll/src/main/c/io_netty_channel_unix_FileDescriptor.c

```diff
@@ -38,3 +145,106 @@ JNIEXPORT int JNICALL Java_io_netty_channel_unix_FileDescriptor_open(JNIEnv* env
     }
     return res;
 }
+
+JNIEXPORT jint JNICALL Java_io_netty_channel_unix_FileDescriptor_write(JNIEnv* env, jclass clazz, jint fd, jobject jbuffer, jint pos, jint limit) {
+    void* buffer = (*env)->GetDirectBufferAddress(env, jbuffer);
+    if (buffer == NULL) {
+        netty_unix_errors_throwRuntimeException(env, "failed to get direct buffer address");
+        return -1;
+    }
```

Comment:
I think in this case the user should not use the epoll transport ;) I think it is better to eliminate the check here so we not need to pay on every call.


> On 23 Oct 2015, at 02:09, Scott Mitchell <notifications@github.com> wrote:
> 
> In transport-native-epoll/src/main/c/io_netty_channel_unix_FileDescriptor.c <https://github.com/netty/netty/pull/4335#discussion_r42821689>:
> 
> > @@ -38,3 +145,106 @@ JNIEXPORT int JNICALL Java_io_netty_channel_unix_FileDescriptor_open(JNIEnv* env
> >      }
> >      return res;
> >  }
> > +
> > +JNIEXPORT jint JNICALL Java_io_netty_channel_unix_FileDescriptor_write(JNIEnv* env, jclass clazz, jint fd, jobject jbuffer, jint pos, jint limit) {
> > +    void* buffer = (*env)->GetDirectBufferAddress(env, jbuffer);
> > +    if (buffer == NULL) {
> > +        netty_unix_errors_throwRuntimeException(env, "failed to get direct buffer address");
> > +        return -1;
> > +    }
> I guess it is possible a user's JVM doesn't support this but the user doesn't call this method...however I'm not sure how likely this is.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/netty/netty/pull/4335/files#r42821689>.
>